docs/Design/SoftDetailedDes/MIS.tex

```
@@ -642,6 +646,106 @@ \subsubsection{Local Functions}
 
 N/A
 
+% MIS of Job Management Module Beginning
+\section{MIS of Job Management Module} \label{Module}
+
+\subsection{Module}
+
+Module for contractors to enter their job information.
+
+\subsection{Uses}
+boto3 (AWS SDK for Python), Database Interaction Module
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+N/A
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+  \begin{tabular}{p{3cm} p{4cm} p{2cm} p{3cm}}
+    \hline
+    \textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+    \hline
+    createJob & \textbf{jobType:}string & boolean & None \\
+    addWorkOrder & \textbf{workOrder:}int & boolean &
+    illegalArgument: Work Order not valid number \\
+    addDescription & \textbf{description:}string & boolean & None \\
+    uploadFile & \textbf{file}: any file type & boolean &
```

A bit difficult to explain, but in AWS you can have something called a presigned URL, which basically has credentials built into a URL which you can make API calls against. Instead of having the backend manage uploading files, which has strict limitations in file size (10 mb), we instead are gonna provide a presigned URL to the UI and have the UI manage the upload (which has a limit of 5 GB or 5 TB depending on how you do it). So basically this should return a URL. 

The input here is also a little odd, I think it should also be taking a unique identifier of the job, and the name of the file being uploaded, and since the UI is doing the uploading remove the file.  

We also don't really need the boolean output, the function should raise an exception if any failure occurs

---

Here I don't even think uploadFile is needed or any interactivity with the File Storage module. The UI can request for the presigned url directly and pass it to our API to be stored in the database.

---

Yeah okay, although the presigned url doesn't actually give any indication of the s3 location its being uploaded to I don't think, but we can always just return that in the API call that's requesting the presigned URL

---

Ok sounds good, I'll remove the uploadFile method from this module